File: neural java/Connection.java
package itu.ejuu.neural;

public class Connection {
	
	private Node fromNode, toNode;
	private double weight;

	public Connection(double weight, Node fromNode, Node toNode){
		this.weight = weight;
		this.fromNode = fromNode;
		this.toNode = toNode;
		
		// adding to nodes
		fromNode.conn_out.add(this);
		toNode.conn_in.add(this);
	}
	
	public double getWeight(){
		return weight;
	}
	
	public void changeWeight(double amount){
		weight += amount;
	}
	
	public Node getFromNode(){
		return fromNode;
	}
	
	public Node getToNode(){
		return toNode;
	}

	public void updateWeight() {
		double delta = Node.L*toNode.getError()*fromNode.getOutput();
		this.changeWeight(delta);
	}
}

File: DataMining/Neural Networks/Connection.cs
namespace DataMining.Neural_Networks
{
    public class Connection
    {
        public Node From { get; private set; }
        public Node To { get; private set; }
        public double Weight{ get; private set; }

        public Connection(double weight, Node fromNode, Node toNode)
        {
            Weight = weight;
            From = fromNode;
            To = toNode;

            From.OutCon.Add(this);
            To.InCon.Add(this);
        }

        public void ChangeWeight(double change)
        {
            Weight += change;
        }

        public void UpdateWeight()
        {
            double d = Node.LearningRate*To.ErrorValue*From.OutputValue;
            ChangeWeight(d);
        }
    }
}